src/mjolnir/graphvalidator.cc

```diff
@@ -556,7 +556,8 @@ void GraphValidator::Validate(const boost::property_tree::ptree& pt) {
   for (const auto& id : tileset) {
     tilequeue.emplace_back(id);
   }
-  std::random_shuffle(tilequeue.begin(), tilequeue.end());
+  // fixate seed for reproducible tile build
+  std::shuffle(tilequeue.begin(), tilequeue.end(), std::mt19937(3));
```

Comment:
utrecht tiles build use concurrency=1 so it wasn't the main cause. However, the fixating seed still makes sense imho

Reply:
i cant understand how fixing the seed doesnt resolve the issue. how is the order of the bin indeterminant if only 1 thread is doing the work?

Reply:
yea, turned out `Validate` section looked at the wrong `concurrency` field. Now it takes `mjolnir.concurrency` => fixated seed solves the issue

Reply:
btw, seed also helps with single-thread `reproduce_tile_build`-test because `random_shuffle` uses `rand()` under the hood(I guess) so different builds of the test use different shuffles in that case
So we might think about setting seed for rand function in the very begging of the tile building
```cpp
// e.g

bool build_tile_set(...) {
  srand(3); // fixated magic seed for rand() function
  ...
```

Reply:
@merkispavel that makes sense but... that has the effect of being global which isnt good for unit tests, i think what you have here is good where you pass the generator to shuffle